### PR TITLE
NAS-125806 / 24.04 / Disable restricted controls for Shares and Data Protection pages

### PR DESCRIPTION
**Pull request by bvasilenko**
**Testing**

1. Create a new local group on our server, with `READONLY` role and another group with `SHARING_MANAGER` role at **Credentials > Local Groups > Privileges**.

2. Log in with this user, and visit the **Shares** page, click through every button and clickable icon. 
   - **Expected result:** if action is illogical, or corresponding API call results in an error, then this action should not be available in WebUI for the current user

5. Do the same for **Data Protection** page.

### CODE REVIEW

**Review by AlexKarpov98 — CHANGES_REQUESTED** [review]
Let's fix all of the 🔴 RED checks first: 👇 

Screenshot 2023-12-26 at 15 11 22

File: src/app/pages/services/components/service-smb/service-smb.component.ts [hunk 1]
```diff
@@ -60,6 +61,7 @@ export class ServiceSmbComponent implements OnInit {
     multichannel: [false, []],
   });
 
+  readonly requiresRoles = [Role.SharingNfsWrite, Role.SharingManager, Role.SharingWrite];
```
Comment (denysbutenko) [inline src/app/pages/services/components/service-smb/service-smb.component.ts:64]: `Role.SharingNfsWrite` seems like a wrong role here

**Review by AlexKarpov98 — APPROVED** [review]
It works as expected. 👍

**Review by denysbutenko — APPROVED** [review]
👍🏻

**Pull request merged**